### PR TITLE
Add smoke test Using a simple RN50 with torch.compile

### PR DESCRIPTION
**Pull request by atalman**
Add smoke test based on: https://github.com/pytorch/builder/issues/1318

### CODE REVIEW

**Review by NicolasHug — APPROVED** [review]
Stamping to unblock but @pmeier comments seem relevant

File: test/smoke_test.py [hunk 1]
```diff
@@ -26,6 +26,13 @@ def smoke_test_torchvision_read_decode() -> None:
     if img_png.ndim != 3 or img_png.numel() < 100:
         raise RuntimeError(f"Unexpected shape of img_png: {img_png.shape}")
 
+def smoke_test_compile() -> None:
+    import torch.nn as nn
+    model = resnet50().cuda()
+    model = torch.compile(model)
+    x = torch.randn(1, 3, 224, 224).cuda()
+    out = model(x)
+    print(out.shape)
```
[the remark this answers is not in the document]
Comment (atalman) [reply test/smoke_test.py]: @pmeier We basically want to see if compile works in this PR. As per: https://github.com/pytorch/builder/issues/1318
Will do more testing of this and perhaps add assert as you suggest in the next PR.

**Comment by NicolasHug** [comment]
@atalman should we cherry-pick that into the release branch immediately, or do we need to made for a new RC to be cut first?

**Comment by atalman** [comment]
@NicolasHug no not necessary for cherry pick since we are running tests from main.

**Pull request merged**